packages/atclient/src/atsign.c

```diff
@@ -1,13 +1,37 @@
-#ifndef ATCLIENT_ATSIGN_H
-#define ATCLIENT_ATSIGN_H
-
 #include "atclient/atsign.h"
 #include "atlogger/atlogger.h"
 #include <stdio.h>
 #include <string.h>
 
 #define TAG "atsign"
 
+int atclient_atsign_init(atclient_atsign *atsign, const char *atsign_str) {
+  int ret = 0;
+
+  // atsign_str is longer than expected or null/empty
+  if ((strlen(atsign_str) > 56) || (atsign_str == NULL) || (strlen(atsign_str) == 0)) {
+    ret = 1;
+    atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atclient_atsign_init: %d\n", ret);
+    return ret;
+  }
+
+  memset(atsign, 0, sizeof(atclient_atsign));
+  atsign->atsign = malloc(strlen(atsign_str) + 1);
+
+  const unsigned long maxatlen = 56;
```

Comment:
I think this may need to be 57? 1 for @, 55 for name part, 1 for null term. Or is the `@` included in the 55 count?

Reply:
Okay, so apparently the maximum length is 55 utf7 characters, after the utf8 (single char) string is converted to utf8 for persistence

Reply:
https://github.com/atsign-foundation/at_c/pull/111/commits/29c2134259a2e506a74eb6db84438f5c2f588e2d